cmd/collectors/zapiperf/plugins/workload/workload.go

```diff
@@ -0,0 +1,438 @@
+
+	lunParams := &LunWorkloadParams{
+		data:           data,
+		qosPolicyMap:   qospolicyMap,
+		lunWorkloadMap: lunWorkloadMap,
+		volumeNames:    lunVolumeNames,
+		svmNames:       lunSvmNames,
+	}
+	err = w.populateUsedIOPsPercentLun(lunParams)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return nil, w.client.Metadata, nil
+}
+
+type VolumeWorkloadParams struct {
+	data              *matrix.Matrix
+	qosPolicyMap      map[string]*collectors.QosAdaptive
+	volumeWorkloadMap map[string]string
+	volumeNames       *set.Set
+	svmNames          *set.Set
+}
+
+func (w *Workload) populateUsedIOPsPercentVolume(params *VolumeWorkloadParams) error {
+	var (
+		result  *node.Node
+		volumes []*node.Node
+		err     error
+	)
+
+	if params.volumeNames.Size() == 0 {
+		return nil
+	}
+
+	query := "volume-get-iter"
+	tag := "initial"
+	request := node.NewXMLS(query)
+	request.NewChildS("max-records", batchSize)
+	desired := node.NewXMLS("desired-attributes")
+	volumeAttributes := node.NewXMLS("volume-attributes")
+	volumeSpaceAttributes := node.NewXMLS("volume-space-attributes")
+	volumeSpaceAttributes.NewChildS("size", "")
+	volumeSpaceAttributes.NewChildS("size-used", "")
+	volumeAttributes.AddChild(volumeSpaceAttributes)
+
+	volumeIDAttributes := node.NewXMLS("volume-id-attributes")
+	volumeIDAttributes.NewChildS("name", "")
+	volumeIDAttributes.NewChildS("owning-vserver-name", "")
+	volumeAttributes.AddChild(volumeIDAttributes)
+	desired.AddChild(volumeAttributes)
+
+	queryAttribute := node.NewXMLS("query")
+	volumeQueryAttributes := node.NewXMLS("volume-attributes")
+	volumeIDQueryAttributes := node.NewXMLS("volume-id-attributes")
+	volumeIDQueryAttributes.NewChildS("name", strings.Join(params.volumeNames.Values(), "|"))
+	volumeIDQueryAttributes.NewChildS("owning-vserver-name", strings.Join(params.svmNames.Values(), "|"))
+	volumeQueryAttributes.AddChild(volumeIDQueryAttributes)
+	queryAttribute.AddChild(volumeQueryAttributes)
+
+	request.AddChild(queryAttribute)
+	request.AddChild(desired)
+
+	for {
+		if result, tag, err = w.client.InvokeBatchRequest(request, tag, ""); err != nil {
+			return err
+		}
+
+		if result == nil {
+			break
+		}
+
+		if x := result.GetChildS("attributes-list"); x != nil {
+			volumes = x.GetChildren()
+		}
+		if len(volumes) == 0 {
+			return nil
+		}
+
+		for _, v := range volumes {
+			size := v.GetChildS("volume-space-attributes").GetChildContentS("size")
+			sizeUsed := v.GetChildS("volume-space-attributes").GetChildContentS("size-used")
+			volume := v.GetChildS("volume-id-attributes").GetChildContentS("name")
+			svm := v.GetChildS("volume-id-attributes").GetChildContentS("owning-vserver-name")
+			key := volume + token + svm
+			instance := params.data.GetInstance(params.volumeWorkloadMap[key])
+			if instance != nil {
+				err = w.processWorkloadInstance(instance, size, sizeUsed, params.qosPolicyMap, params.data)
+				if err != nil {
+					w.Logger.Error().Err(err).Str(key, key).Msg("Failed process workload")
```

Comment:
Should key be quoted like so?
`w.Logger.Error().Err(err).Str("key", key).Msg("Failed process workload")`